test/Chainweb/Test/Utils.hs

```diff
@@ -649,20 +659,68 @@ withTestAppServer tls v appIO envIO userFunc = bracket start stop go
     go (_, _, env) = userFunc env
 
 
+data ValidationException = ValidationException
+    { vReq :: W.Request
+    , vResp :: (ResponseHeaders, Status, BL.ByteString)
+    , vErr :: WV.TopLevelError
+    }
+    deriving (Show, Typeable)
+
+instance Exception ValidationException
+
+{-# NOINLINE chainwebOpenApiSpec #-}
+chainwebOpenApiSpec :: OpenApi
+chainwebOpenApiSpec = unsafePerformIO $ do
+    mgr <- manager 10_000_000
+    let specUri = "https://raw.githubusercontent.com/kadena-io/chainweb-openapi/main/chainweb.openapi.yaml"
+    Yaml.decodeThrow . BL.toStrict . HTTP.responseBody =<< HTTP.httpLbs (HTTP.parseRequest_ specUri) mgr
+
+{-# NOINLINE pactOpenApiSpec #-}
+pactOpenApiSpec :: OpenApi
+pactOpenApiSpec = unsafePerformIO $ do
+    mgr <- manager 10_000_000
+    let specUri = "https://raw.githubusercontent.com/kadena-io/chainweb-openapi/main/pact.openapi.yaml"
+    Yaml.decodeThrow . BL.toStrict . HTTP.responseBody =<< HTTP.httpLbs (HTTP.parseRequest_ specUri) mgr
+
 -- TODO: catch, wrap, and forward exceptions from chainwebApplication
 --
 withChainwebTestServer
     :: Bool
+    -> Bool
     -> ChainwebVersion
     -> IO W.Application
     -> (Int -> IO a)
     -> (IO a -> TestTree)
     -> TestTree
-withChainwebTestServer tls v appIO envIO test = withResource start stop $ \x ->
+withChainwebTestServer validateSpec tls v appIO envIO test = withResource start stop $ \x -> do
     test $ x >>= \(_, _, env) -> return env
   where
     start = do
-        app <- appIO
+        coverageRef <- newIORef $ WV.CoverageMap Map.empty
+        _ <- evaluate chainwebOpenApiSpec
+        _ <- evaluate pactOpenApiSpec
+        let
+            lg (_, req) (respBody, resp) err = do
+                let ex = ValidationException req (W.responseHeaders resp, W.responseStatus resp, respBody) err
+                print $ ppShow ex
+                error "validation error"
+            findPath path = asum
+                [ case B8.split '/' path of
```

Comment:
This code feels duplicated from the code in the other module. Perhaps it should be extracted to some kind of `Util` function?